src/libsyntax/ext/expand.rs

```diff
@@ -72,6 +71,20 @@ macro_rules! ast_fragments {
         }
 
         impl AstFragment {
+            pub fn add_placeholders(&mut self, placeholders: &[NodeId]) {
+                if placeholders.is_empty() {
+                    return;
+                }
+                match self {
+                    $($(AstFragment::$Kind(ast) => ast.extend(placeholders.iter().flat_map(|id| {
+                        #[cfg_attr(bootstrap, allow(unused_macros))]
+                        macro _repeating($flat_map_ast_elt) {}
```

Comment:
To mention the `$flat_map_ast_elt` token somehow.
Otherwise we get `error: attempted to repeat an expression containing no syntax variables matched as repeating at this depth` because the outer `macro_rules! ast_fragments` doesn't know what to repeat here.
Any suggestions on how to name `_repeating` in a more clear way, or should I just add a comment?